src/js/atomShuffle.js

```diff
@@ -1,86 +1,59 @@
-(function ($) {
-  $.fn.atomShuffleInstance = function (customClass) {
-    const className =
+class atomShuffleInstance {
+  constructor(el, customClass = null) {
+    if (!(el instanceof HTMLElement)) {
+      console.warn("AtomShuffleInstance: Element is not a valid HTMLElement");
+      return null;
+    }
+    var className =
       customClass && customClass instanceof String
-        ? "." + customClass
-        : ".shufflingItem";
-
-    this.initialize = function () {
-      $(this).keyup(function () {
-        var input = $(this).val().toLowerCase();
-        if (input) {
-          $(className).each(function () {
-            $(this).is(":hidden") &&
-              $(this).attr("data-title").toLowerCase().includes(input) &&
-              $(this).show(400);
-            $(this).is(":visible") &&
-              !$(this).attr("data-title").toLowerCase().includes(input) &&
-              $(this).hide(300);
-          });
+        ? customClass
+        : "atomShuffleItem";
+    el.addEventListener("keyup", function () {
+      var val = el.value.toLowerCase();
+      var items = document.getElementsByClassName(className);
+      Array.from(items).forEach((it) => {
+        if (val) {
+          !a_is_visible(it) &&
+            it.dataset.title.toLowerCase().includes(val) &&
+            a_show(it);
+          a_is_visible(it) &&
+            !it.dataset.title.toLowerCase().includes(val) &&
+            a_hide(it);
```

Comment:
```suggestion
          if(!a_is_visible(it) && it.dataset.title.toLowerCase().includes(val))
            a_show(it);
          if(a_is_visible(it) && !it.dataset.title.toLowerCase().includes(val))
            a_hide(it);
```
Or checking in `a_show` and `a_hide` directly